soundfile.py

```diff
@@ -167,6 +167,38 @@
     __libsndfile_version__ = __libsndfile_version__[len('libsndfile-'):]
 
 
+class SoundFileError(Exception):
+    """Base class for all SoundFile-specific errors."""
+    pass
+
+class SoundFileRuntimeError(SoundFileError, RuntimeError):
+    """SoundFile runtime error.
+
+    Errors that used to be `RuntimeError`."""
+    pass
+
+class LibsndfileError(SoundFileRuntimeError):
+    """libsndfile errors.
+
+
+    Attributes
+    ----------
+    code
+        libsndfile internal error number
+
+    error_string
+        raw libsndfile error message
+    """
+    def __init__(self, code, prefix=""):
+        SoundFileRuntimeError.__init__(self, code, prefix)
```

Comment:
Why is this necessary?

I thought we don't need to call the base class constructor ...?

Reply:
This won't hurt anything and will make args work as expected in Python 2.7… even though no one should care about Python 2 any more. In fact lack of calling base class constructor can be considered an error by various coding standard or code validation tools. I won't insist on this, though

Reply:
I think we should keep Python 2 compatibility for the time being. We can discuss elsewhere if it is time to drop compatibility for v1.